get-artifacts/README.md

```diff
@@ -0,0 +1,60 @@
+# Get Artifacts #
+
+A downloader for artifacts from multiple prior workflow_runs (by workflow_file_name and branch), or a single run by workflow_run_id.
+
+Artifacts are always uploaded as zips, correspondingly this action allows you to unzip and then delete the original downloads with the "decompress" flags.
+
+If an ```${{ inputs.target_dir }}/<workflow_run_id>/<artifact_name>/``` folder already exist no attempt will be made to download the artifacts for that run.  The assumption it github's cache action can be use to preserve prior downloads between workflow executions, preventing unnecessary downloads.  More info on cacheing [here](https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows "caching-dependencies-to-speed-up-workflows").
+
+## Examples ##
+
+Download artifacts from one workflow run.
+
+```yaml
+  test-get-artifacts-single:
+    runs-on: ubuntu-latest
+    name: test get-artifacts from a single workflow run
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download and extract single run's artifacts.
+        uses: ./get-artifacts
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          workflow_run_id: ${{ github.event.workflow_run.id }}
+          artifacts: test.download
+          # the following two lines are default values included for clarity.
+          target_dir: ${{ github.workspace }}/downloads
+          decompress: true
+      - name: Test that "test.download" got downloaded
+        run: |
+          [[ -f  ${{ github.workspace }}/downloads/${{ github.event.workflow_run.id }}/test.download/test-file.download ]]
+          [ $( ls ${{ github.workspace }}/downloads/ | wc -l ) = 1 ]
+      - name: Test that "test.not-download" did not get downloaded
+        run: |
+          [[ ! -f  ${{ github.workspace }}/downloads/${{ github.event.workflow_run.id }}/test.not-download/test-file.not-download ]]
+```
```

Comment:
Could the example be expanded to include caching?

Reply:
My mid term plan for the shell in this action is to use it in another action that does everything (and assumes locations/naming patterns for unit test artifacts and file names).   I wound't expect most projects to need to call this action directly.   While overkill is still dead, I don't want to build a full flow in the examples.   No problem copying them out of diem when it's used there, before test locations are fully formalized.

Reply:
Ahh I see. That makes sense.